common.go

```diff
@@ -0,0 +1,110 @@
+package cose
+
+import (
+	"errors"
+	"fmt"
+)
+
+// IntOrStr is a value that can be
+type IntOrStr struct {
+	intVal *int
+	strVal *string
+}
+
+func NewIntOrStr(v interface{}) *IntOrStr {
+	var ios IntOrStr
+	if err := ios.Set(v); err != nil {
+		return nil
+	}
+	return &ios
+}
+
+func (ios IntOrStr) Int() int {
+	if ios.intVal != nil {
+		return *ios.intVal
+	}
+	return 0
```

Comment:
Is this return tied to the fact that 0 is the invalid algorithm?  If so, it seems more appropriate to rename `IntOrStr` to `COSEAlgorithm` (or similar), to make the underlying assumptions clearer.